src/rsa/signing.rs

```diff
@@ -77,8 +77,29 @@ impl RSAKeyPair {
     /// version of *ring* will likely replace the support for the
     /// `RSAPrivateKey` format with support for the PKCS#8 format.
     ///
+    /// The private key is validated according to [NIST SP-800-56B rev. 1]
+    /// section 6.4.1.4.3, crt_pkv (Intended Exponent-Creation Method Unknown),
+    /// with the following exceptions:
+    /// - Section 6.4.1.2.1, Step 1: Neither a target security level nor an
+    ///   expected modulus length is provided as a parameter, so checks
+    ///   regarding these expectations are not done.
+    /// - Section 6.4.1.2.1, Step 3: Since neither the public key nor the
+    ///   expected modulus length is provided as a parameter, the consistency
+    ///   check between these values and the private key's value of n isn't done.
+    /// - Section 6.4.1.2.1, Step 5: No primality tests are done, both for
+    ///   performance reasons and to avoid any side channels that such tests
+    ///   would provide.
+    ///
+    /// Steps 2 and 4 as described in 6.4.1.2.1 are omitted per 6.4.1.4.3.
+    ///
+    /// In addition to the NIST requirements, we require that p > q and that
+    /// e < 2^33.
```

Comment:
I suggest instead of "e < 2^33" that we either use "e < 2**33" or say that "e must be no more than 33 bits."